cypress/e2e/settings.cy.js

```diff
@@ -1,32 +1,51 @@
 /// <reference types="cypress" />
 /// <reference types="../support" />
 
+import SettingPageObject from '../support/pages/settingsPageObject';
+
+const settingsPage = new SettingPageObject();
+
 describe('Settings page', () => {
-  before(() => {
+  let user;
 
+  before(() => {
+    cy.task('generateUser').then((generateUser) => {
+      user = generateUser;
+    });
   });
 
   beforeEach(() => {
-
+    cy.task('db:clear');
+    cy.visit('/');
+    cy.login();
+    settingsPage.visit();
   });
 
   it('should provide an ability to update username', () => {
-
+    settingsPage.typeUsername(user.username);
+    settingsPage.clickUpdateSettingsBtn();
+    settingsPage.assertProfilePage(user.username);
   });
 
   it('should provide an ability to update bio', () => {
-
+    settingsPage.typeBio(user.bio);
+    settingsPage.clickUpdateSettingsBtn();
   });
 
   it('should provide an ability to update an email', () => {
+    settingsPage.typeEmail(user.email);
```

Comment:
The `typeEmail` method should ensure that the email field is cleared before typing the new email. This is important to avoid appending to any existing text in the field.